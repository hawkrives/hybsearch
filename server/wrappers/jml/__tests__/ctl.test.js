/* eslint-env jest */

const generateControlFile = require('../ctl')
const fs = require('fs')
const path = require('path')

const base = path.join(__dirname, 'input')
const files = fs.readdirSync(base).filter(f => f.endsWith('.phy'))

for (const file of files) {
	test(file, () => {
		const content = fs.readFileSync(path.join(base, file), 'utf-8')
		const tree = generateControlFile(content)
		expect(tree).toMatchSnapshot()
	})
}
