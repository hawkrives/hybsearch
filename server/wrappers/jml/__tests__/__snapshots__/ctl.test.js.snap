// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emydura-short.phy 1`] = `
"species = tQwNDLi oTYwLH vwAqcZ CXTpAGBdz
seqperspecies = 3 2 2 1

locusrate = 1.0
heredityscalar = 0.25
seqgencommand = -mHKY -f0.2678,0.1604,0.2031,0.3687 -t1.5161 -i0 -a0.2195 -l810
significancelevel = 0.9999
burnin = 0
thinning = 1
seed = -1
"
`;

exports[`smallerlepus.phy 1`] = `
"species = igcfean abbdbkga cbcidpnh bkaabphc
seqperspecies = 15 2 8 4

locusrate = 1.0
heredityscalar = 0.25
seqgencommand = -mHKY -f0.2678,0.1604,0.2031,0.3687 -t1.5161 -i0 -a0.2195 -l810
significancelevel = 0.9999
burnin = 0
thinning = 1
seed = -1
"
`;
